sound/soc/sof/ipc4-topology.c

```diff
@@ -497,7 +529,17 @@ static int sof_ipc4_widget_setup_msg(struct snd_sof_widget *swidget, struct sof_
 
 	msg->extension = SOF_IPC4_MOD_EXT_CORE_ID(swidget->core);
 
-	type = (fw_module->man4_module_entry.type & SOF_IPC4_MODULE_DP) ? 1 : 0;
+	switch (swidget->comp_domain) {
+	case SOF_COMP_DOMAIN_LL:
+		type = 0;
+		break;
+	case SOF_COMP_DOMAIN_DP:
+		type = 1;
+		break;
+	default:
```

Comment:
now that enum values align with IPC values and default unset is LL we can just do the assignment to IPC and remove this.

Reply:
We can not rely on IPC value alone. If IPC is not set we should follow the manifest value. I added comments to the enum values to make it clear why they are needed.